### PR TITLE
ArticulatedObjects: Enable support for secondary json config files.

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR introduces support of Json config files for URDF-based ArticulatedObject models in the URDFParser, which will hold default user-defined attributes (Such as bounding boxes for placements) that apply to all instances of articulated objects built from these models.   This PR also removed extraneous markers from error messages that did not match the format of our current message logging in the URDFParser.

To be compatible with this functionality, the json config file holding the user_defined attributes must meet these criteria : 
- Must be in the same directory as the AO's .urdf file
- Must have the same core filename (minus extensions) as the AO's .urdf file
- Must have the following extension :  <filename>**.ao_config.json**
- Internally, the file must have a root Json object, and within that object there must be a tag "user_defined" which references a Json object holding the user-defined values.

## How Has This Been Tested
Locally c++ and python tests.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/physics/PhysicsManager.cpp [hunk 1]
```diff
@@ -304,9 +304,9 @@ int PhysicsManager::addArticulatedObjectInstance(
   existingArticulatedObjects_.at(aObjID)->setSceneInstanceAttr(
       aObjInstAttributes);
 
-  // set articulated object's user-defined attributes, if any exist in scene
+  // merge articulated object's user-defined attributes, if any exist in scene
   // instance.
-  existingArticulatedObjects_.at(aObjID)->setUserAttributes(
+  existingArticulatedObjects_.at(aObjID)->mergeUserAttributes(
```
Comment (Skylion007) [inline src/esp/physics/PhysicsManager.cpp:309]: Nit: calling at(aObjId) is not free. The reference should really be cached between these calls, don't you think?

File: src/esp/io/URDFParser.cpp [hunk 2]
```diff
@@ -185,13 +240,13 @@ bool Parser::parseURDF(std::shared_ptr<Model>& urdfModel,
         urdfModel->m_links[link->m_name] = link;
       }
     } else {
-      ESP_ERROR() << "E - failed to parse link. Aborting parse/load for"
+      ESP_ERROR() << "Failed to parse link. Aborting parse/load for"
                   << filename;
       return false;
     }
   }
   if (urdfModel->m_links.size() == 0) {
```
Comment (Skylion007) [inline src/esp/io/URDFParser.cpp:248]:
```suggestion
  if (urdfModel->m_links.empty()) {
```

**Review by Skylion007 — APPROVED** [review]

File: src/esp/physics/PhysicsManager.cpp (hunk 1 above)
Comment (jturner65) [reply src/esp/physics/PhysicsManager.cpp:309]: Every little bit helps, yeah

**Review by aclegg3 — APPROVED** [review]
Looks great, thanks for cleaning up. 👍

**Pull request merged**